lib/src/node/compile.dart

```diff
@@ -227,29 +227,25 @@ Importer _parseImporter(Object? importer) {
 
 /// Implements the simplification algorithm for custom function return values.
 /// {@link https://github.com/sass/sass/blob/main/spec/types/calculation.md#simplifying-a-calculationvalue}
-dynamic simplify(dynamic value) {
-  if (value is SassCalculation) {
-    var simplifiedArgs = value.arguments.map(simplify).toList().cast<Object>();
-    if (value.name == 'calc') {
-      return simplifiedArgs[0];
-    }
-    if (value.name == 'clamp') {
-      return simplify(Function.apply(SassCalculation.clamp, simplifiedArgs));
-    }
-    if (value.name == 'min') {
-      return simplify(Function.apply(SassCalculation.min, [simplifiedArgs]));
-    }
-    if (value.name == 'max') {
-      return simplify(Function.apply(SassCalculation.max, [simplifiedArgs]));
-    }
-    return SassCalculation.unsimplified(value.name, simplifiedArgs);
-  }
-  if (value is CalculationOperation) {
-    return simplify(SassCalculation.operate(value.operator,
-        simplify(value.left) as Object, simplify(value.right) as Object));
-  }
-  return value;
-}
+Object simplify(Object value) => switch (value) {
+      SassCalculation() => switch ((
+          // Match against...
+          value.name, // ...the calculation name
+          value.arguments.map(simplify).toList() // ...and simplified arguments
+        )) {
+          ('calc', [var first, ...]) => first,
```

Comment:
As with `clamp`, this should be

```suggestion
          ('calc', [var first]) => first,
```

and then have a fallback case that throws an error for more arguments.